Add files via upload

Seshan01 opened the pull request:
Pull request to add initial schemas for the entities - Tank, offGasStack, Junction and Blower. More entities are required to needed to be added in the incubated repository.
Initial base schema of the data model has also been uploaded.

Seshan01 committed:
Add files via upload

albertoabellagarcia commented on wastewater/dataModel.WasteWaterTreatment/Junction/schema.json:
```diff
@@ -0,0 +1,75 @@
+{
```
We have a problem with the name of this data model because it [already exists](https://github.com/smart-data-models/dataModel.WaterNetworkManagement/tree/master/Junction) a junction in waterNetwrokmanagement subject. Please review it and propose a marge (in the case is the same concept) o an alternative name. 

It looks to be quite different, maybe TreatmentJunction could be an alternative

albertoabellagarcia commented on wastewater/dataModel.WasteWaterTreatment/Tank/schema.json:
```diff
@@ -0,0 +1,75 @@
+{
```
Same comment as junction

albertoabellagarcia commented on wastewater/dataModel.WasteWaterTreatment/WasteWaterTreatment-schema.json:
```diff
@@ -0,0 +1,760 @@
+                "value"
+            ]
+        },
+        "ngsildRelationship": {
```
not necessary (it should be removed)

albertoabellagarcia commented on wastewater/dataModel.WasteWaterTreatment/WasteWaterTreatment-schema.json:
```diff
@@ -0,0 +1,760 @@
+    "description": "Common NGSI-LD definitions for Wastewater Treatment data models",
+    "type": "object",
+    "definitions": {
+        "ngsildProperty": {
```
not necessary (it should be removed)

albertoabellagarcia commented on wastewater/dataModel.WasteWaterTreatment/WasteWaterTreatment-schema.json:
```diff
@@ -0,0 +1,760 @@
+                "value"
+            ]
+        },
+        "NH4": {
```
Subproperties defined in the NGSI LD standard are not necessary (i.e. unitcode, observedAt, etc) it should be removed

albertoabellagarcia commented on wastewater/dataModel.WasteWaterTreatment/WasteWaterTreatment-schema.json:
```diff
@@ -0,0 +1,760 @@
+                "value"
+            ]
+        },
+        "NO3": {
```
Subproperties defined in the NGSI LD standard are not necessary (i.e. unitcode, observedAt, etc) it should be removed

albertoabellagarcia commented on wastewater/dataModel.WasteWaterTreatment/WasteWaterTreatment-schema.json:
```diff
@@ -0,0 +1,760 @@
+                "value"
+            ]
+        },
+        "DO": {
```
Subproperties defined in the NGSI LD standard are not necessary (i.e. unitcode, observedAt, etc) it should be removed

albertoabellagarcia commented on wastewater/dataModel.WasteWaterTreatment/WasteWaterTreatment-schema.json:
```diff
@@ -0,0 +1,760 @@
+                "value"
+            ]
+        },
+        "Redox": {
```
Subproperties defined in the NGSI LD standard are not necessary (i.e. unitcode, observedAt, etc) it should be removed

albertoabellagarcia commented on wastewater/dataModel.WasteWaterTreatment/WasteWaterTreatment-schema.json:
```diff
@@ -0,0 +1,760 @@
+                "value"
+            ]
+        },
+        "Sludge level": {
```
Subproperties defined in the NGSI LD standard are not necessary (i.e. unitcode, observedAt, etc) it should be removed

albertoabellagarcia commented on wastewater/dataModel.WasteWaterTreatment/WasteWaterTreatment-schema.json:
```diff
@@ -0,0 +1,760 @@
+                "value"
+            ]
+        },
+        "TN": {
```
Subproperties defined in the NGSI LD standard are not necessary (i.e. unitcode, observedAt, etc) it should be removed

albertoabellagarcia commented on wastewater/dataModel.WasteWaterTreatment/WasteWaterTreatment-schema.json:
```diff
@@ -0,0 +1,760 @@
+{
```
globally it would be good to check if these properties already exist in the [database of properties](https://smartdatamodels.org/index.php/ddbb-of-properties-descriptions/ ) it could be referenced to an existing schema, or reallocated in shared schema (like this file)

Seshan01 replied:
Sure! Is it possible to see an example as to how I can reference a property to an existing schema?

albertoabellagarcia replied:
[An example](https://github.com/smart-data-models/dataModel.TourismDestinations/blob/master/TouristTrip/schema.json) from tourism.
if a property is used in several data models it should be located in :
- In a shared file at subject level 'subject'-schema.json (i.e. [waternetworkmanagemet](https://github.com/smart-data-models/dataModel.WaterNetworkManagement/blob/master/WaterNetworkManagement-schema.json))
- In a shared file at domain level 'domain'-schema.json
- In a shared file at global level like [common-schema.json](https://github.com/smart-data-models/data-models/blob/master/common-schema.json)
What option should you chose? it depends on the data models that are going to share the property.

Seshan01 replied:
Thanks Alberto! This is helpful. As we are also interested in connecting parameters to devices (sensors or actuators), it would make sense to therefore also refer to the Device and DeviceModel. Say for example we are interested in the dateLastCalibration property. Can we simply do the following:

"dateLastCalibration": {
         "$ref": "https://smart-data-models.github.io/dataModel.Device/Device/schema.json#/properties/dateLastCalibration"
}
or do we need to specify it in $ref with "https://smart-data-models.github.io/dataModel.Device/device-schema.json#/definitions/Device-Commons"?
But as dateLastCalibration (and many other properties) are not part of Device-Commons, I imagine this will not work?

albertoabellagarcia replied:
Both are possible, but the second option would be better.  This could mean a PR also on device repo, not a problem.

albertoabellagarcia commented on wastewater/dataModel.WasteWaterTreatment/WasteWaterTreatment-schema.json:
```diff
@@ -0,0 +1,760 @@
+                "value"
+            ]
+        },
+        "Temperature": {
```
Subproperties defined in the NGSI LD standard are not necessary (i.e. unitcode, observedAt, etc) it should be removed

albertoabellagarcia commented on wastewater/dataModel.WasteWaterTreatment/WasteWaterTreatment-schema.json:
```diff
@@ -0,0 +1,760 @@
+                "value"
+            ]
+        },
+        "pH": {
```
Subproperties defined in the NGSI LD standard are not necessary (i.e. unitcode, observedAt, etc) it should be removed

albertoabellagarcia commented on wastewater/dataModel.WasteWaterTreatment/WasteWaterTreatment-schema.json:
```diff
@@ -0,0 +1,760 @@
+                "value"
+            ]
+        },
+        "airflow": {
```
Subproperties defined in the NGSI LD standard are not necessary (i.e. unitcode, observedAt, etc) it should be removed

albertoabellagarcia commented on wastewater/dataModel.WasteWaterTreatment/WasteWaterTreatment-schema.json:
```diff
@@ -0,0 +1,760 @@
+                "value"
+            ]
+        },
+        "energy": {
```
Subproperties defined in the NGSI LD standard are not necessary (i.e. unitcode, observedAt, etc) it should be removed

albertoabellagarcia commented on wastewater/dataModel.WasteWaterTreatment/WasteWaterTreatment-schema.json:
```diff
@@ -0,0 +1,760 @@
+                "value"
+            ]
+        },
+        "pressure": {
```
Subproperties defined in the NGSI LD standard are not necessary (i.e. unitcode, observedAt, etc) it should be removed

albertoabellagarcia commented on wastewater/dataModel.WasteWaterTreatment/WasteWaterTreatment-schema.json:
```diff
@@ -0,0 +1,760 @@
+                "value"
+            ]
+        },
+        "airflowEstimation": {
```
Subproperties defined in the NGSI LD standard are not necessary (i.e. unitcode, observedAt, etc) it should be removed

albertoabellagarcia commented on wastewater/dataModel.WasteWaterTreatment/WasteWaterTreatment-schema.json:
```diff
@@ -0,0 +1,760 @@
+                "value"
+            ]
+        },
+        "status": {
```
Subproperties defined in the NGSI LD standard are not necessary (i.e. unitcode, observedAt, etc) it should be removed

albertoabellagarcia commented on wastewater/dataModel.WasteWaterTreatment/WasteWaterTreatment-schema.json:
```diff
@@ -0,0 +1,760 @@
+                "value"
+            ]
+        },
+        "tag": {
```
Subproperties defined in the NGSI LD standard are not necessary (i.e. unitcode, observedAt, etc) it should be removed

albertoabellagarcia commented on wastewater/dataModel.WasteWaterTreatment/WasteWaterTreatment-schema.json:
```diff
@@ -0,0 +1,760 @@
+                "value"
+            ]
+        },
+        "emissionFlow": {
```
Subproperties defined in the NGSI LD standard are not necessary (i.e. unitcode, observedAt, etc) it should be removed

albertoabellagarcia commented on wastewater/dataModel.WasteWaterTreatment/WasteWaterTreatment-schema.json:
```diff
@@ -0,0 +1,760 @@
+                "value"
+            ]
+        },
+        "TOC": {
```
Subproperties defined in the NGSI LD standard are not necessary (i.e. unitcode, observedAt, etc) it should be removed

albertoabellagarcia commented on wastewater/dataModel.WasteWaterTreatment/WasteWaterTreatment-schema.json:
```diff
@@ -0,0 +1,760 @@
+                "value"
+            ]
+        },
+        "TIC": {
```
Subproperties defined in the NGSI LD standard are not necessary (i.e. unitcode, observedAt, etc) it should be removed. be explicit with meaning of TIC

albertoabellagarcia commented on wastewater/dataModel.WasteWaterTreatment/WasteWaterTreatment-schema.json:
```diff
@@ -0,0 +1,760 @@
+                "type",
+                "value"
+            ]
+        },
```
Subproperties defined in the NGSI LD standard are not necessary (i.e. unitcode, observedAt, etc) it should be removed

albertoabellagarcia commented on wastewater/dataModel.WasteWaterTreatment/WasteWaterTreatment-schema.json:
```diff
@@ -0,0 +1,760 @@
+                "value"
+            ]
+        },
+        "BOD": {
```
Subproperties defined in the NGSI LD standard are not necessary (i.e. unitcode, observedAt, etc) it should be removed

albertoabellagarcia commented on wastewater/dataModel.WasteWaterTreatment/WasteWaterTreatment-schema.json:
```diff
@@ -0,0 +1,760 @@
+                "value"
+            ]
+        },
+        "COD": {
```
Subproperties defined in the NGSI LD standard are not necessary (i.e. unitcode, observedAt, etc) it should be removed

albertoabellagarcia commented on wastewater/dataModel.WasteWaterTreatment/WasteWaterTreatment-schema.json:
```diff
@@ -0,0 +1,760 @@
+                "value"
+            ]
+        },
+        "Flowrate": {
```
Subproperties defined in the NGSI LD standard are not necessary (i.e. unitcode, observedAt, etc) it should be removed

albertoabellagarcia commented on wastewater/dataModel.WasteWaterTreatment/WasteWaterTreatment-schema.json:
```diff
@@ -0,0 +1,760 @@
+                "value"
+            ]
+        },
+        "N2O": {
```
Subproperties defined in the NGSI LD standard are not necessary (i.e. unitcode, observedAt, etc) it should be removed

albertoabellagarcia commented on wastewater/dataModel.WasteWaterTreatment/WasteWaterTreatment-schema.json:
```diff
@@ -0,0 +1,760 @@
+                "value"
+            ]
+        },
+        "CO2": {
```
Subproperties defined in the NGSI LD standard are not necessary (i.e. unitcode, observedAt, etc) it should be removed

albertoabellagarcia commented on wastewater/dataModel.WasteWaterTreatment/WasteWaterTreatment-schema.json:
```diff
@@ -0,0 +1,760 @@
+                "value"
+            ]
+        },
+        "CH4": {
```
Subproperties defined in the NGSI LD standard are not necessary (i.e. unitcode, observedAt, etc) it should be removed

albertoabellagarcia commented on wastewater/dataModel.WasteWaterTreatment/WasteWaterTreatment-schema.json:
```diff
@@ -0,0 +1,760 @@
+                "value"
+            ]
+        },
+        "O2": {
```
Subproperties defined in the NGSI LD standard are not necessary (i.e. unitcode, observedAt, etc) it should be removed

albertoabellagarcia commented on wastewater/dataModel.WasteWaterTreatment/offGasStack/schema.json:
```diff
@@ -0,0 +1,55 @@
+            "$ref": "https://smart-data-models.github.io/data-models/common-schema.json#/definitions/Location-Commons"
+        },
+        {
+            "property": {
```
it has to be properties.

albertoabellagarcia commented:
Please review the comments and apply. 
Additionally some example payloads are necessary.
Please use the [contribution manual](https://bit.ly/contribution_manual) for further details

Seshan01 committed:
Committing changes as reviewed by Alberto through pull request comments.
Name of the Tank and Junction entities have been changed to WasteWaterTank and WasteWaterJunction.
Removal of sub-properties in the base schema.
Fixing of type error in OffGasStack entity.

Seshan01 committed:
Committing minor changes in the Blower and OffGasStack schemas.
Committing a new schema for the WasteWaterDevice. Most properties are common to Device model. New properties to be added to support the wastewater use case.

Seshan01 committed:
Committing minor change in the WasteWaterJunction schema.

albertoabellagarcia committed:
Merge pull request #38 from Seshan01/Seshan01-KWR-1

Committing changes as reviewed by Alberto through pull request comments.
